packages/adminrouter/extra/src/lib/cache.lua

```diff
@@ -255,15 +255,21 @@ local function fetch_and_store_marathon_apps(auth_token)
        -- In "container/bridge" and "host" networking modes we need to use the
        -- host port for routing (available via task's ports array)
        local port
-       if is_container_network(app) and app["container"]["portMappings"][portIdx]["containerPort"] ~= 0 then
-         port = app["container"]["portMappings"][portIdx]["containerPort"]
-       else
-         port = task["ports"][portIdx]
+       if is_container_network(app) then
+           if not app["container"]["portMappings"][portIdx] then
+               ngx.log(ngx.NOTICE, "Cannot find port in container portMappings at Marathon port index '" .. (portIdx - 1) .. "' for app '" .. appId .. "'")
```

Comment:
Why did you choose `NOTICE`? (not implying anything else for now, just would love to understand how you thought about it!)

Reply:
Is it helpful to have `portIdx` in code and `portIdx - 1` in the log message?

Reply:
can we shorten this line?

Reply:
Can you quickly summarize in which cases `if not app["container"]["portMappings"][portIdx]` is true? What happens if `app["container"]` does not have the property `["portMappings"]`?

Reply:
I see it as in general there are two log levels of interest: NOTICE (INFO) and DEBUG.
Since we don't halt on this (ERROR case) and  but rather just skip and continuing cache refreshing, emitting a NOTICE is the reasonable thing to do. One could make it a WARNING, but from the perspective of Nginx nothing went wrong. However the user should get a reasonable response on what went wrong and we the deployment didn't work (at the Marathon level?). This should not be done primarily in the Admin Router logs IMO.

Reply:
Continue cases:
1. `container` networking mode, `portMappings` defined, `portMappings` list index out of bounds.
2. `container/bridge` networking mode, `ports` list index out of bounds.
3. `host` networking mode, `ports` list index out of bounds.

> What happens if app["container"] does not have the property ["portMappings"]
Then, the first part is skipped without notice and the ports array index port is used instead.
I'm not sure what's the intended behavior here.